src/history/hs.c

```diff
@@ -1088,7 +1099,16 @@ __wt_hs_cursor_position(WT_SESSION_IMPL *session, WT_CURSOR *cursor, uint32_t bt
             WT_STAT_DATA_INCR(session, cursor_skip_hs_cur_position);
 
             WT_ERR(__wt_compare(session, NULL, &cursor->key, srch_key, &cmp));
```

Comment:
Should this __wt_compare use the key retrieved from search_near (i.e: hs_key on the line below), instead of the key passed in for the search?

Reply:
I think that would change the functionality provided by this function making it no longer place on the end of the updates for a given btree id?